homeassistant/components/reolink/store.py

```diff
@@ -0,0 +1,73 @@
+"""Local storage for the Reolink integration."""
+
+import asyncio
+import logging
+from pathlib import Path
+
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ReolinkStore:
+    """Local storage for Reolink."""
+
+    def __init__(self, hass: HomeAssistant, config_id: str) -> None:
+        """Initialize ReolinkStore."""
+        self._hass = hass
+        self._path = Path(hass.config.path(f".storage/{DOMAIN}/{config_id}.json"))
+        self._lock = asyncio.Lock()
+
+    async def async_load(self) -> str:
+        """Load the data from disk."""
+        async with self._lock:
+            return await self._hass.async_add_executor_job(self._load)
+
+    def _load(self) -> str:
+        """Load the data from disk."""
+        if not self._path.exists():
+            _LOGGER.debug("Failed to load file %s: path does not exist", self._path)
+            return "{}"
+
+        try:
+            return self._path.read_text(encoding="utf-8")
+        except OSError as err:
+            _LOGGER.debug("Failed to load file %s: %s", self._path, err)
+            return "{}"
+
+    async def async_store(self, data: str) -> None:
```

Comment:
HAs version has a delayed save, which is a callback. That way you don't need to wait until the data is stored

Reply:
But then the retrieval of the data also needs to be a seperate function right?